friends/src/components/FriendEditor.js

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+
+export default function FriendEditor({ form, addFriend, updateFriend, isEditing, inputChange }) {
+	const onNameChange = event => {
+		inputChange(event.target.value, 'nameValue');
+	};
+
+	const onEmailChange = event => {
+		inputChange(event.target.value, 'emailValue');
+	};
+
+	const onAgeChange = event => {
+		inputChange(event.target.value, 'ageValue');
+	};
+
+	const onFriendAdd = event => {
+		addFriend();
+	};
+
+	const onFriendUpdate = event => {
+		updateFriend();
+	};
+
```

Comment:
after adding or updating a friend you could redirect to the friends page, have a look at the ```<Redirect />``` component from ```react-router-dom```